src/server/v2.0/handler/project.go

```diff
@@ -557,6 +560,18 @@ func (a *projectAPI) UpdateProject(ctx context.Context, params operation.UpdateP
 		log.Warningf("failed to call JSONCopy on project metadata when UpdateProject, error: %v", err)
 	}
 
+	// validate retention_id
+	if ridParam, ok := p.Metadata["retention_id"]; ok {
+		md, err := a.metadataMgr.Get(ctx, p.ProjectID)
+		if err != nil {
+			return a.SendError(ctx, err)
+		}
+		if rid, ok := md["retention_id"]; !ok || rid != ridParam {
```

Comment:
Replace 
```
if rid, ok := md["retention_id"]; !ok || rid != ridParam
```
With 
```
if ridParam != md["retention_id"]
```
will make it simpler.

Reply:
it cannot be replace that way. if ridParam is passed in as an empty string and this project doesn't have any retention policy,  we should send a 400 code, while `if ridParam != md["retention_id"]` cannot achieve that.